async-nats/src/jetstream/consumer/push.rs

```diff
@@ -255,3 +255,218 @@ impl IntoConsumerConfig for &Config {
+                            }
+                            Poll::Pending => {
+                                return Poll::Pending;
+                            }
+                        }
+                    }
+                    Some(subscriber) => match subscriber.as_mut().poll(cx) {
+                        Poll::Ready(subscriber) => {
+                            self.subscriber_future = None;
+                            self.subscriber = Some(subscriber?);
+                        }
+                        Poll::Pending => {
+                            return Poll::Pending;
+                        }
+                    },
+                }
+            }
+            if let Some(subscriber) = self.subscriber.as_mut() {
+                match subscriber.receiver.poll_recv(cx) {
+                    Poll::Ready(maybe_message) => match maybe_message {
+                        Some(message) => match message.status {
+                            Some(StatusCode::IDLE_HEARBEAT) => {
+                                if let Some(subject) = message.reply {
+                                    // TODO store pending_publish as a future and return errors from it
+                                    let client = self.context.client.clone();
+                                    tokio::task::spawn(async move {
+                                        client
+                                            .publish(subject, Bytes::from_static(b""))
+                                            .await
+                                            .unwrap();
+                                    });
+                                }
+                                continue;
+                            }
+                            Some(_) => {
+                                continue;
+                            }
+                            None => {
+                                let jetstream_message = jetstream::message::Message {
+                                    message,
+                                    context: self.context.clone(),
+                                };
+
+                                let info = jetstream_message.jetstream_message_info()?;
+                                // let sequence = info.stream_seq;
+                                if info.stream_seq != self.stream_sequence + 1 {
+                                    self.subscriber = None;
+                                    continue;
+                                }
+                                self.stream_sequence = info.stream_seq;
+                                return Poll::Ready(Some(Ok(jetstream_message)));
+                            }
+                        },
+                        None => return Poll::Ready(None),
+                    },
+                    Poll::Pending => return Poll::Pending,
+                }
+            }
+        }
+    }
+}
+
+async fn recreate_ephemeral_subscriber(
```

Comment:
Maybe throw this into the impl (private)? less parameters to pass.

Reply:
hm, but as private for Iterator? yeah, we can do that.

Reply:
Alright